psl-core/src/main/java/org/linqs/psl/model/rule/AbstractRule.java

```diff
@@ -17,16 +17,29 @@
  */
 package org.linqs.psl.model.rule;
 
+import java.util.HashMap;
 import java.util.List;
+import java.util.Map;
 
 /**
  * Base class for all (first order, i.e., not ground) rules.
  */
 public abstract class AbstractRule implements Rule {
     protected final String name;
 
+    private static Map<Integer, AbstractRule> Rules = new HashMap<Integer, AbstractRule>();
+
+    public static AbstractRule getRule(int hashcode) {
+        return Rules.get(hashcode);
+    }
+
+    public static void addRule(AbstractRule rule) {
```

Comment:
Go ahead and forgo this method, and instead put the line directly in the constructor.
We want this mapping and rule constructor to be very tightly coupled.

Note that in cases where a method like this would exist, it should be private instead of public.